androidApp/src/main/java/com/mbta/tid/mbta_app/android/map/MapLayerManager.kt

```diff
@@ -44,12 +48,56 @@ class MapLayerManager(val map: MapboxMap, context: Context) {
         }
     }
 
-    suspend fun addLayers(colorPalette: ColorPalette) {
-        val layers: List<MapboxLayer> =
-            RouteLayerGenerator.createAllRouteLayers(colorPalette).map { it.toMapbox() } +
-                StopLayerGenerator.createStopLayers(colorPalette).map { it.toMapbox() }
+    suspend fun addLayers(
+        mapFriendlyRouteResponse: MapFriendlyRouteResponse,
+        globalResponse: GlobalResponse,
+        colorPalette: ColorPalette
+    ) {
+        val routeLayers: List<MapboxLayer> =
+            RouteLayerGenerator.createAllRouteLayers(
+                    mapFriendlyRouteResponse,
+                    globalResponse,
+                    colorPalette
+                )
+                .map { it.toMapbox() }
+        val stopLayers = StopLayerGenerator.createStopLayers(colorPalette).map { it.toMapbox() }
+        setLayers(routeLayers, stopLayers)
+    }
+
+    suspend fun addLayers(
+        routes: List<MapFriendlyRouteResponse.RouteWithSegmentedShapes>,
+        globalResponse: GlobalResponse,
+        colorPalette: ColorPalette
+    ) {
+        val routeLayers =
+            RouteLayerGenerator.createAllRouteLayers(routes, globalResponse, colorPalette).map {
+                it.toMapbox()
+            }
+        val stopLayers = StopLayerGenerator.createStopLayers(colorPalette).map { it.toMapbox() }
+        setLayers(routeLayers, stopLayers)
+    }
+
+    private suspend fun setLayers(routeLayers: List<MapboxLayer>, stopLayers: List<MapboxLayer>) {
         withContext(Dispatchers.Main) {
-            for (layer in layers) {
+            if (!map.styleLayerExists(bufferLayerId)) {
+                val layer = SlotLayer(bufferLayerId)
```

Comment:
Inserting an empty layer with a known ID between the route layers and the stop layers prevents routes from being added on top of stops, which was happening before I did that, but it doesn’t technically guarantee that individual routes won’t be drawn in the wrong order if one already existed and another one is being added later. If we decide it’s worth sinking unbounded engineer time into route sorting on the map, we could probably loop back through after we’ve added all the layers to make sure that everything is in the order it’s supposed to be, but I’d just as soon not deal with that.

Reply:
If there are routes that we draw at the same time that run close enough to overlap, we might draw the wrong one on top if you’re transitioning from a state that only draws the one that should be higher to a state that draws both. Also, at intersections between routes, we might draw the wrong route on top, but that doesn’t matter.